### PR TITLE
Feature/replace rscript

### PR DESCRIPTION
**Pull request by laestrada**
Replace Lu's R script with a python translated version with the following updates:
- no 30ppb cap on the bias adjustment
- replace infilling of nan values with the mean bias of the entire time period to instead use a 30-day moving mean

### CODE REVIEW

File: Step3_correct_background/calculate_bias.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# between the two using the latitudinal averages and 
+# spatial/temporal smoothing. 
+# This script is based off of Lu Shen's 
+# methods/ R script.
+#######################################################
+
+# settings to adjust default smoothing windows
+# we use a +/- 15 day time window and 3x3
+# smoothing for lat/lon
+smoothing_lat_window = 3
```
Comment (djvaron) [inline Step3_correct_background/calculate_bias.py:15]: So is this a 12 x 15 degree lat/lon window? I thought we used a 20 x 20 degree window (10-degree radius), which would be 5 x 4 grid cells.

File: Step3_correct_background/calculate_bias.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+smoothing_lon_window = 3
+smoothing_time_window = 30
+
+# access the preprocessed CH4 data
+filepath = "Daily_CH4.nc"
+daily_CH4 = xr.open_dataset(filepath)
+TROPOMI_lon = daily_CH4["lon"]
+TROPOMI_lat = daily_CH4["lat"]
+GC_CH4 = daily_CH4["GC"]
+OMI_CH4 = daily_CH4["CH4"]
```
Comment (djvaron) [inline Step3_correct_background/calculate_bias.py]: Should this variable be renamed `TROPOMI_CH4` for clarity?

File: Step3_correct_background/calculate_bias.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+GC_bkgd = smooth_3D_da(GC_CH4)
+TROPOMI_bkgd = smooth_3D_da(OMI_CH4)
+
+# calculate bias between GC background CH4 and
+# TROPOMI observational background CH4
+bias_4x5 = GC_bkgd - TROPOMI_bkgd
+
+# build a smoothed dataset to fill in nan values of raw bias
+# start by smoothing the bias
+bias_avg_base = smooth_3D_da(bias_4x5)
```
Comment (djvaron) [inline Step3_correct_background/calculate_bias.py]: Why do we smooth `bias_4x5`? It's the difference between two already smoothed fields. Just checking.

File: Step3_correct_background/calculate_bias.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+for i in range(len(TROPOMI_lon)):
+    lat_base_full[:, :, i] = lat_base
+
+# infill nan values in smoothed bias with latitudinal averages
+# for each corresponding day and smooth the result
+bias_avg_base = smooth_3D_da(bias_avg_base.fillna(lat_base_full))
+
+# infill nan values of raw bias data with the smoothed
+# average background data and smooth the final product
+bias_4x5_new = smooth_3D_da(bias_4x5.fillna(bias_avg_base))
```
Comment (djvaron) [inline Step3_correct_background/calculate_bias.py]: Here and Line 84 we call `smooth_3D_da` again. I'm just curious why we need to smooth the fields so many times. The final output field seems to have seen `smooth_3D_da` four times?

File: Step3_correct_background/calculate_bias.py (hunk 2 above)
Comment (laestrada) [reply Step3_correct_background/calculate_bias.py]: Yes, good catch!

File: Step3_correct_background/calculate_bias.py (hunk 3 above)
Comment (laestrada) [reply Step3_correct_background/calculate_bias.py]: This one is an error -- removed!

**Comment by laestrada** [comment]
All comments are resolved. I will hold off on merging until we have completed a comparison between the old R script and new python script outputs.

**Pull request merged**